hack/deploy/configure-values.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script interrogates the AKS cluster and Azure resources to generate 
+# the karpenter-values.yaml file using the karpenter-values-template.yaml file as a template.
+
+# Check the cluster name and resource group are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <cluster-name> <resource-group>"
+    exit 1
+fi
+
+echo "Configuring karpenter-values.yaml for cluster $1 in resource group $2 ..."
+
+AZURE_CLUSTER_NAME=$1
+AZURE_RESOURCE_GROUP=$2
+
+AZURE_LOCATION=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".location")
+AZURE_RESOURCE_GROUP_MC=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".nodeResourceGroup")
+
+KARPENTER_SERVICE_ACCOUNT_NAME=karpenter-sa
+AZURE_KARPENTER_USER_ASSIGNED_IDENTITY_NAME=karpentermsi
+
+AZURE_CLIENT_ID=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".identityProfile.kubeletidentity.clientId")
```

Comment:
This shouldn't be needed if we're using a `workload` identity. 

I hadn't updated the `savm` path, but it should be updated, and the need for this should be removed.

Reply:
Good catch, removed